src/pngimage.cpp

```diff
@@ -461,11 +461,12 @@ namespace Exiv2 {
                 } else if (chunkType == "iCCP") {
                     // The ICC profile name can vary from 1-79 characters.
                     uint32_t iccOffset = 0;
-                    while (iccOffset < 80 && iccOffset < chunkLength) {
-                         if (chunkData.pData_[iccOffset++] ==  0x00) {
-                            break;
-                         }
-                    }
+                    do {
+                      if (iccOffset >= 80 || iccOffset >= chunkLength) {
+                        throw Exiv2::Error(kerFailedToReadImageData);
```

Comment:
Any particular reason why you have inserted this line? If I understand your patch correctly, then the actual fix is to replace the the while loop with a do-while loop and swap the conditions.

Reply:
The bug happens when `chunkLength == 0`. The old loop would terminate immediately with `iccOffset == 0`. But the code after the loop assumes that `iccOffset > 0` because it is supposed to be the length of the string including the `'\0'` terminator. If you replaced this `throw` with a `break`, then the bug would still happen.

Reply:
Unfortunately, I don't really know the details of the PNG format, so let's keep it this way.